calyx/ir/src/builder.rs

```diff
@@ -60,6 +60,43 @@ impl<'a> Builder<'a> {
         self.component.continuous_assignments.extend(assigns);
     }
 
+    /// Construct a non-conflicting name for the FSM construct
+    pub fn generate_fsm_name(&mut self) -> ir::Id {
```

Comment:
How often is this function used within the code? If the only use of this function is within the implementation of the new pass, I would rather put this in there as opposed to exposing this within the Builder API.